generated/nixnetsocket/nixnetsocket.proto

```diff
@@ -49,6 +50,30 @@ message SockAddr {
   }
 }
 
+enum OptName {
+  OPT_NAME_UNSPECIFIED = 0;
+  OPT_NAME_TCP_NODELAY = 1;
+  OPT_NAME_IP_MULTICAST_TTL = 6;
+  OPT_NAME_IPV6_MULTICAST_HOPS = 15;
+  OPT_NAME_SO_RX_DATA = 257;
+  OPT_NAME_SO_RCV_BUF = 258;
+  OPT_NAME_SO_SND_BUF = 259;
+  OPT_NAME_SO_NON_BLOCK = 260;
+  OPT_NAME_SO_BIND_TO_DEVICE = 261;
+  OPT_NAME_SO_ERROR = 262;
+  OPT_NAME_SO_LINGER = 263;
+  OPT_NAME_SO_REUSE_ADDR = 264;
+}
+
+message SockOptData {
+  OptName opt_name = 1;
```

Comment:
Check for understanding: The reason for having opt_name in the struct is to make it easier to implement the converter right? So we have all the data we need from a single grpc type? I think that's OK. it seems like mildly-negative-to-neutral from a client perspective.

I think we talked about this a little bit speculating that maybe doing this allows us to pair the valid value with the correct type in the type system. Like a tagged union in typescript. But I don't think that `protobuf` gives us a way to do that, at least not using enums. So the behavior is the same as if you separated the name from the data.

Reply:
Actually, I don't know if it made it easier to implement the converter. I thought it was nice to have them coupled but I don't think that's true. it makes more sense having SockOptData just hold the oneof data and have the user use OptName enum for the optname input.

Yeah, I think that was the initial reason I had them together but it dosn't make sense in the current state. Now SockOptData just has the oneof data field and the user inputs a separate field on the request for the opt name.